spec/aruba/spawn_process_spec.rb

```diff
@@ -56,5 +56,35 @@
 
       it { expect {process.start}.to raise_error Aruba::LaunchError }
     end
+
+    context "with a childprocess launch error" do
+      let(:child) { instance_double(ChildProcess::AbstractProcess) }
```

Comment:
No, that won't work. Or I don't see how.

A class double would help if I was only mocking out `ChildProcess.build`. And since `Aruba::Processes::SpawnProcess` breaks the Law Of Demeter here (the result of `.build` should be wrapped by another Aruba class), I can't use a `const_double` for it. 

I mean, I can used `const_double` for `.build` because technically it "fits" (as a method on a constant), but that's not what I'm testing. If I did, I'd be forced to use `:transfer_nested_constants` because everything in the tests is namespaced with `ChildProcess`. And that wouldn't make sense, because `ChildProcess` is the ONLY case here of using a class method.

As for "fiddling with someone else's classes" - that's the point of using `instance_double` in the first place - to make the implementation based on the interface (of the 3rd party class) while being independent from the behavior. Thus, "mocking".

It may seem debatable whether using `ChildProcess::AbstractProcess` is a good idea, but I think it's the best option anyway. Even if I used `ChildProcess::Windows::Process` instead (if it was available without requiring implicitly), it would still be the exact same interface (due to inheritance). And because of inheritance, `ChildProcess::AbstractProcess` is the public API of `childprocess` anyway, so that won't change without a major bump.

All I really wanted to do is mock out the `:start` method, but due to the way the class is setup (and not wrapped by another Aruba class), the way the mocks are done here just exposes a design problem already in codebase (not wrapping the 3rd party class with something easier to mock out).

I can't use `object_double` on the child, because then the test would depend on the signature of `ChildProcess::AbstractProcess#initialize` (which *is* a private API - the only method of the class!).

So it's either terrible mocking here, or changing the Aruba API to wrap the `ChildProcess.build` result.

I can use a double, but that's not any less verbose ... and ... it's would just be way of completely ignoring the API altogether.

I can't think of a better alternative here than adding a new wrapper class in Aruba.